tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/reindexing/ElasticsearchGetTagsTests.java

```diff
@@ -0,0 +1,216 @@
+            .withCoreAPI()
+            .getTags()
+            .assertThat()
+            .entrySetContains("tag", orange.getTag().toLowerCase()));
+    }
+
+    @AfterClass
+    public void dataCleanup()
+    {
+        STEP("Remove created tags");
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().usingTag(apple).deleteTag();
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().usingTag(banana).deleteTag();
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().usingTag(pineapple).deleteTag();
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().usingTag(winegrape).deleteTag();
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().usingTag(grapefruit).deleteTag();
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().usingTag(orange).deleteTag();
+
+        STEP("Remove site and user");
+        dataSite.usingUser(user).deleteSite(site);
+        dataUser.deleteUser(user);
+    }
+
+    /**
+     * Verify if exact name filter can be applied.
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.TAGS, TestGroup.REGRESSION })
+    public void testGetTags_withSingleNameFilter()
+    {
+        STEP("Get tags with names filter using EQUALS and expect one item in result");
+        final RestTagModelsCollection returnedCollection = restClient.authenticateUser(user)
+            .withParams("where=(tag='" + apple.getTag() + "')")
+            .withCoreAPI()
+            .getTags();
+
+        restClient.assertStatusCodeIs(HttpStatus.OK);
+        returnedCollection.assertThat()
+            .entrySetMatches("tag", Set.of(apple.getTag().toLowerCase()));
+    }
+
+    /**
+     * Verify if multiple names can be applied as a filter.
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.TAGS, TestGroup.REGRESSION })
+    public void testGetTags_withTwoNameFilters()
+    {
+        STEP("Get tags with names filter using IN and expect two items in result");
+        final RestTagModelsCollection returnedCollection = restClient.authenticateUser(user)
+            .withParams("where=(tag IN ('" + apple.getTag() + "', '" + banana.getTag() + "'))")
+            .withCoreAPI()
+            .getTags();
+
+        restClient.assertStatusCodeIs(HttpStatus.OK);
+        returnedCollection.assertThat()
+            .entrySetMatches("tag", Set.of(apple.getTag().toLowerCase(), banana.getTag().toLowerCase()));
+    }
+
+    /**
+     * Verify if alike name filter can be applied.
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.TAGS, TestGroup.REGRESSION })
+    public void testGetTags_whichNamesStartsWithOrphan()
+    {
+        STEP("Get tags with names filter using MATCHES and expect one item in result");
+        final RestTagModelsCollection returnedCollection = restClient.authenticateUser(user)
+            .withParams("where=(tag MATCHES ('*an*'))")
```

Comment:
I would use more characters than this, and include some of the randomly generated ones, otherwise it seems like there's a good chance that "an" will be present in another tag at some point.

Reply:
This is why I'm checking here if result set contains expected tags, and not matches them.